### PR TITLE
45880472 VQ-VAE

### PR DESCRIPTION
**Pull request by yousifpatti**
Dear Shakes and teaching staff,

Please review this pull request of my implementation of the VQ-VAE model. The files which were added as part of this development are:

driver.py: driver file for the model
model.py: VQ-VAE, encoder, decoder, and Pixel CNN, implemented in Tensorflow.
data_loader.py: functions to load the data and normalise.
README.md: a README file detailing the model architecture, loss function, and results of training.
outputs/ a directory containing images used throughout README.md
Please let me know if you require me to make any changes to this model. Thank you!

Yousif Al-Patti (45880472)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
Difficulty Hard

Functionality
- VQ not quit working
- Missing prior sampling.

Code
- Structure: OK
- Commenting: OK


README 
- Content: missing usage

PR Traget OK

General Comments
- Commit messages: some could be more descriptive
- Commit content: remove some images. We only need a few for demo.

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation, not quite working correctly -2
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description OK
